api/envoy/config/core/v3/protocol.proto

```diff
@@ -157,11 +157,10 @@ message HttpProtocolOptions {
 
   // The maximum duration of a connection. The duration is defined as a period since a connection
   // was established. If not set, there is no max duration. When max_connection_duration is reached
-  // and if there are no active streams, the connection will be closed. If there are any active streams,
-  // the drain sequence will kick-in, and the connection will be force-closed after the drain period.
-  // See :ref:`drain_timeout
+  // and if there are no active streams, the connection will be closed. If the connection is an HTTP/2
```

Comment:
why the change?  AFIK the drain sequence kicks in for HTTP/1 too, which is sending connection:close on responses and force closing.

Reply:
I was reading the docs and code for idle_timeout and felt this matched. It was also important to call out that the drain sequence won't kick in for upstream connections, only downstream ones.

Reply:
Why don't you think it applies to HTTP/1 and HTTP/3 though?

Reply:
I looked at the implementation and noted a call to `goAway()` which likely includes H3 but it made me think that it didn't apply to H1 (but, mostly because of the existing docs on idle_timeout).

The addition of "HTTP/2 downstream connections" instead of just "downstream connections" is possibly an overreach. The main point from my change is that upstream connections won't be subject to a drain_timeout.

Reply:
For example, onConnectionDurationTimeout() and onIdleTimeout both startDrainSequence() and call codec_->shutdownNotice() (appears to no-op for H1) and later, on drain timeout, codec_->goAway() (also appears to no-op for H1).

But now that I read this more, I see that the subsequent call to checkForDeferredClose() is likely responsible for sending Connection: close downstream.

So now I'm curious why the idle_timeout docs are worded the way they are. Maybe there right thing to do is simply remove the "HTTP/2" distinction from idle_timeout since it appears that draining happens for all. Thoughts?

Reply:
yeah I'd be inclined to remove the HTTP/2
again while the shutdownNotice doesn't do anything for HTTP/1.1 the transition to draining results in connection: close which is the http/1.1 style drain, so I think the sequence of drain followed by force-close works for http 1.1, 2, and 3, just in different ways.